sympy/core/mul.py

```diff
@@ -1880,10 +1880,12 @@ def coeff_exp(term, x):
         try:
             for t in self.args:
                 coeff, exp = t.leadterm(x)
-                if not coeff.has(x):
-                    ords.append((t, exp))
+                if (isinstance(coeff, Expr) or coeff.is_number) \
+                    and not (coeff.has(x) or coeff.has(Abs) or coeff.has(sign)) \
+                    and not (coeff.is_Symbol and not t.is_Symbol):
```

Comment:
This test looks complicated. Can you explain the purpose of its various conditions?

Reply:
Alright, so the first condition is useless, I will remove that.
`if not coeff.has(x)` condition was already present
`if not (coeff.has(Abs) or coeff.has(sign))` Is to fix https://github.com/sympy/sympy/issues/9449, basically Abs and sign containing expressions can be computed, and should not raise `ValueError`
`if not coeff.is_Symbol` is to solve https://github.com/sympy/sympy/pull/20724, however, this ends up breaking
```py
def test_bug4():                                                                
    w = Symbol("w")                                                             
    e = x/(w**4 + x**2*w**4 + 2*x*w**4)*w**4                                    
    assert e.nseries(w, n=2).removeO() in [x/(1 + 2*x + x**2),                  
        1/(1 + x/2 + 1/x/2)/2, 1/x/(1 + 2/x + x**(-2))]                         
```
(0 is returned)
Based on printing the involved variables and seeing the differences in the outputs (hit and trial), I found the condition
 `not (coeff.is_Symbol and not t.is_Symbol)` which is a workaround to allow this test to pass too.


Honestly, a lot of this is just workarounds and a much better, simpler condition is `if coeff.is_number`. The only problem with this condition is outlined in https://github.com/sympy/sympy/issues/20897 and I have mentioned a possible fix in the comment. @jksuom If you could look into that once and tell if that is fine and can be included in this PR, it would reduce the condition to just this simple check and would be much better IMO.

Reply:
It is possible to continue with this method as such - 
```diff
diff --git a/sympy/core/mul.py b/sympy/core/mul.py
index 9b7b113e13..5fbb0125d2 100644
--- a/sympy/core/mul.py
+++ b/sympy/core/mul.py
@@ -1880,12 +1880,11 @@ def coeff_exp(term, x):
         try:
             for t in self.args:
                 coeff, exp = t.leadterm(x)
-                if (isinstance(coeff, Expr) or coeff.is_number) \
-                    and not (coeff.has(x) or coeff.has(Abs) or coeff.has(sign)) \
-                    and not (coeff.is_Symbol and not t.is_Symbol):
+                if not (coeff.has(x) or coeff.is_Symbol) \
+                   and not (coeff.has(Abs) or coeff.has(sign)):
                         ords.append((t, exp))
                 else:
                        raise ValueError
@@ -377,7 +378,7 @@ def test_series3():
 def test_bug4():
     w = Symbol("w")
     e = x/(w**4 + x**2*w**4 + 2*x*w**4)*w**4
-    assert e.nseries(w, n=2).removeO() in [x/(1 + 2*x + x**2),
+    assert e.nseries(w).removeO() in [x/(1 + 2*x + x**2),
         1/(1 + x/2 + 1/x/2)/2, 1/x/(1 + 2/x + x**(-2))]
 
 
 ```
 But I think the new commit I pushed, which simplifies the condition much more and fixes another bug (?) in power, is better.